app/src/main/java/com/example/sharewhatyoucanproject/login/LoginViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package com.example.sharewhatyoucanproject.login
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.DEFAULT_PASSWORD
+import com.example.sharewhatyoucanproject.utils.getRandomName
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.UserProfileChangeRequest
+import com.google.firebase.firestore.FirebaseFirestore
+
+class LoginViewModel(
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance(),
+    private val db: FirebaseFirestore = FirebaseFirestore.getInstance(),
+
+) : ViewModel() {
+
+    val checkResult = MutableLiveData<String>()
+    val loginResult = MutableLiveData<String>()
+    val createUserResult = MutableLiveData<String>()
+    val saveUserResult = MutableLiveData<String>()
+
+    var type: UserType = UserType.UNKNOWN
+    var deviceId: String = ""
+    private var userId: String = ""
+    private var userEmail: String = ""
+
+    fun checkUser(email: String) {
+        db.collection("users")
+            .whereEqualTo("email", email)
+            .get()
+            .addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                    val size = task.result.size()
+                    if (size == 0) {
+                        signUp(email)
+                    } else {
+                        loginUser(email)
```

Comment:
if the user is already logged in and their data already exists, is it necessary to re-login? I'm not familiar with firebase auth so not sure if this is required